libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -188,6 +199,34 @@ size_t ContractCompiler::packIntoContractCreator(ContractDefinition const& _cont
 	return m_context.runtimeSub();
 }
 
+size_t ContractCompiler::deployLibrary(ContractDefinition const& _contract)
+{
+	solAssert(!!m_runtimeCompiler, "");
+	solAssert(_contract.isLibrary(), "Tried to deploy contract as library.");
+
+	CompilerContext::LocationSetter locationSetter(m_context, _contract);
+
+	solAssert(m_context.runtimeSub() != size_t(-1), "Runtime sub not registered");
+	m_context.pushSubroutineSize(m_context.runtimeSub());
+	m_context.pushSubroutineOffset(m_context.runtimeSub());
+	m_context.appendInlineAssembly(R"(
+	{
+		// If code starts at 11, an mstore(0) writes to the full PUSH20 plus data
+		// without the need for a shift.
+		let codepos := 11
+		codecopy(codepos, subOffset, subSize)
+		// Check that the first opcode is a PUSH20
+		switch eq(0x73, byte(0, mload(codepos)))
```

Comment:
You're adding a runtime check to verify that the optimiser isn't broken? That seems... really bad.

Reply:
It's not a runtime check that he optimzer is broken. It is a runtime check that the optimizer did not modify the code beyond what we expect.

Reply:
Is there a difference?

Reply:
Yes, the optimizer is working fine if the resulting code is semantically equivalent, but here, we need a stronger promise: The placeholder where we insert the address has to be at a certain position in the bytecode.